documentation/kernel-semantics.tex

```diff
@@ -3152,12 +3175,29 @@ \subsubsection{Function Declaration}
 \subsection{Event Loop}
 \label{subsec:eventloop}
 
+\dart{} supports asynchronous execution of code with async functions and await expressions.
+We introduce an event loop to define the behaviour of asynchronous code.
+We denote the event loop with $\eventloop$.
+$\eventloop$ represents a list of event continuations, elements of $\dncont$.
+\[
+    \eventloop \in \mlist{\dncont}
+\]
+
+The event loop is a list and it supports the operations defined in Section~\ref{subsubsec:semantic-list}.
+
+The event loop appears on the left and the right-hand side of all configurations, therefore we omit it from rules where the same queue just passed from one state to the other.
+We assume it is implicit and we explicitly mention the cases when the event loop on the right-hand side is not the same as the one passed in the next state.
 
 \subsubsection{Async}
 \label{subsubsec:async}
 
+Asynchronous execution in \dart{} is supported with async functions or asynchronous for in loops.
+In \kernel{} await is an expression and here we present the behaviour of invocation of asynchronous functions.
+Asynchronous for-in loops occur as statements.
+The semantics of the await expression in \kernel{} for expressions other then the above mentioned does not modify the event loop.
```

Comment:
Modified it to say "expression that do not evaluate to values in domain FutureValue".